pynng/nng.py

```diff
@@ -492,6 +492,61 @@ def remove_post_pipe_remove_cb(self, callback):
         """
         self._on_post_pipe_remove.remove(callback)
 
+    def _get_pipe_from_msg(self, msg):
+        lib_pipe = lib.nng_msg_get_pipe(msg._nng_msg)
+        pipe_id = lib.nng_pipe_id(lib_pipe)
+        if pipe_id < 0:
+            # TODO: Better exception
+            raise Exception('No such pipe')
+        pipe = self._pipes[pipe_id]
+        return pipe
+
+    def recv_msg(self, block=True):
+        """
+        Return a Message object.
+
+        """
+        flags = 0
+        if not block:
+            flags |= lib.NNG_FLAG_NONBLOCK
+        msg_p = ffi.new('nng_msg **')
+        check_err(lib.nng_recvmsg(self.socket, msg_p, flags))
+        msg = msg_p[0]
+        msg = Message(msg)
+        msg.pipe = self._get_pipe_from_msg(msg)
+        return msg
+
+    def send_msg(self, msg, block=True):
+        """
+        Send the Message ``msg`` on the socket.
+        """
+        flags = 0
+        if not block:
+            flags |= lib.NNG_FLAG_NONBLOCK
+        with msg._mem_freed_lock:
+            msg._ensure_can_send()
+            check_err(lib.nng_sendmsg(self.socket, msg._nng_msg, flags))
+            msg._mem_freed = True
+
+    async def asend_msg(self, msg):
+        """
+        Asynchronously send the Message ``msg`` on the socket.
+        """
+        with msg._mem_freed_lock:
+            msg._ensure_can_send()
+            with _aio.AIOHelper(self, self._async_backend) as aio:
+                val = await aio.asend_msg(msg)
```

Comment:
`return await aio.asend_msg(msg)`

Reply:
+1, this was cruft leftover from an earlier round of coding.